pymarkdown/plugins/rule_md_050.py

```diff
@@ -0,0 +1,88 @@
+"""
+import re
+from typing import cast
+
+from pymarkdown.inline_markdown_token import ReferenceMarkdownToken
+from pymarkdown.markdown_token import MarkdownToken
+from pymarkdown.plugin_manager.plugin_details import PluginDetails
+from pymarkdown.plugin_manager.plugin_scan_context import PluginScanContext
+from pymarkdown.plugin_manager.rule_plugin import RulePlugin
+
+
+class RuleMd050(RulePlugin):
+    """
+    Class to implement a plugin that validates references based on configurable regexes.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.__regexes = []
+
+    EXTERNAL_LINK_RE = re.compile("^(.+):.*$")
+
+    def get_details(self) -> PluginDetails:
+        """
+        Get the details for the plugin.
+        """
+        return PluginDetails(
+            plugin_name="refs-regex",
+            plugin_id="MD050",
+            plugin_enabled_by_default=False,
+            plugin_description="Reference should match regular expressions",
+            plugin_version="0.0.1",
+            plugin_interface_version=1,
+            plugin_url="https://github.com/jackdewinter/pymarkdown/blob/main/docs/rules/rule_md050.md",
+            plugin_configuration="regexes",
+        )
+
+    def initialize_from_config(self) -> None:
+        """
+        Event to allow the plugin to load configuration information.
+        """
+        self.__regexes = [
+            RegexPair(
+                self.plugin_configuration.get_string_property(f"regexes.{x}.regex"),
+                self.plugin_configuration.get_string_property(
+                    f"regexes.{x}.errorMessage"
+                ),
+            )
+            for x in self.config_indeces()
+        ]
+
+    def config_indeces(self):
+        x = 1
+
+        while (
+            self.plugin_configuration.get_string_property(f"regexes.{x}.regex") != None
+        ):
+            yield x
+            x += 1
+
+    def next_token(self, context: PluginScanContext, token: MarkdownToken) -> None:
+        """
+        Event that a new token is being processed.
+        """
+        if not token.is_inline_image and not token.is_inline_link:
+            return
+        link_uri, _ = self._extract_link_uri(token)
+        for regex in self.__regexes:
+            if not regex.regex.match(link_uri):
+                self.report_next_token_error(
+                    context, token, extra_error_information=regex.error_message
+                )
+
+    def _extract_link_uri(self, token: MarkdownToken):
+        ref_token = cast(ReferenceMarkdownToken, token)
+        link_uri_split = ref_token.link_uri.split("#", 1)
+        if len(link_uri_split) == 2:
+            return link_uri_split
+
+        return link_uri_split[0], None
+
+
+class RegexPair:
+    def __init__(self, regex: str, error_message: str) -> None:
+        self.regex = re.compile(regex)
+        self.error_message = error_message
```

Comment:
Empty line at the end of the file is missing.

Reply:
black removes it for some reason